src/ucp/rma/flush.c

```diff
@@ -444,17 +440,26 @@ static unsigned ucp_worker_flush_progress(void *arg)
     ucs_status_t status;
     ucp_ep_h ep;
 
-    status = ucp_worker_flush_check(worker);
-    if ((status == UCS_OK) || (&next_ep->ep_list == &worker->all_eps)) {
-        /* If all ifaces are flushed, or we finished going over all endpoints,
-         * no need to progress this request actively any more. Just wait until
-         * all associated endpoint flush requests are completed.
-         */
-        ucp_worker_flush_complete_one(req, UCS_OK, 1);
-    } else if (status != UCS_INPROGRESS) {
-        /* Error returned from uct iface flush */
-        ucp_worker_flush_complete_one(req, status, 1);
-    } else if (worker->context->config.ext.flush_worker_eps) {
+    if (worker->flush_ops_count == 0) {
```

Comment:
BTW, do we require iface_flush completion today before destroying UCP worker (AFAIK not)?
since it can leak some endpoints which are still being flushed

Reply:
> BTW, do we require iface_flush completion today before destroying UCP worker (AFAIK not)?

we don't require, but now we require flushing all ops to be done and or iface_flush is completed or going over all EPs is done

Reply:
need to check for leaks in worker destroy

Reply:
done